src/components/Search/SearchComponent.tsx

```diff
@@ -0,0 +1,134 @@
+import { forwardRef, useCallback, useEffect, useImperativeHandle, useRef, useState } from "react";
+import { motion } from "framer-motion";
+import { MagnifyingGlass, X } from "@phosphor-icons/react";
+import { Link, useNavigate } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+
+export type SearchComponentHandler = {
+    toggle: () => void;
+}
+
+export const SearchComponent = forwardRef<SearchComponentHandler>((_, ref) => {
+    const { t } = useTranslation();
+  
+  const [isOpen, setIsOpen] = useState(false);
+  const [historic, setHistoric] = useState<Array<string>>([]);
+  const searchRef = useRef<HTMLInputElement>(null);
+  const backdropRef = useRef<HTMLDivElement>(null);
+  const navigate = useNavigate();
+
+  const storedItems = useRef<Array<string>>(JSON.parse(localStorage.getItem("searchItems") || "[]")).current;
+
+  useImperativeHandle(ref, () => ({
+    toggle: () => {
+      setIsOpen(!isOpen);
+    }
+  }));
+
+  const updateHistoric = useCallback(() => {
+    const currentQuery = searchRef.current?.value;
+
+    if (currentQuery && !historic.includes(currentQuery)) {
+      const newHistoric = [...historic, currentQuery];
+      setHistoric(newHistoric);
+      localStorage.setItem("searchItems", JSON.stringify(newHistoric));
+    }
+  }, [historic]);
+
+  const searchHandler = () => {
+    const searchQuery = searchRef.current?.value;
+
+    if (searchQuery) {
+      updateHistoric();
+      navigate(`/search?query=${encodeURIComponent(searchQuery)}`);
+    }
+  };
+
+  const removeSearchHistoricItem = (item: string) => {
+    const newHistoric = historic.filter(searchItem => searchItem !== item);
+    setHistoric(newHistoric);
+    localStorage.setItem("searchItems", JSON.stringify(newHistoric));
+  }
+
+  const removeAllSearchHistoricItem = () => {
+    setHistoric([]);
+    localStorage.setItem("searchItems", "");
+  }
+
+  useEffect(() => {
+    setHistoric(storedItems);
+  }, []);
+
+  return (
+    <div 
+        ref={backdropRef}
+        className={`fixed inset-0 w-full h-full flex justify-center z-50 ${isOpen ? "" : "pointer-events-none"}`}
+    >
+      <motion.div
+        className="absolute top-0 w-full max-w-[500px] h-dyn-screen bg-black/50 bg-clip-padding backdrop-filter backdrop-blur-sm bg-opacity-10"
+        style={{ opacity: 0, y: -10 }}
+        animate={{ 
+          opacity: isOpen ? 1 : 0,
+          y: isOpen ? 0 : -10
+        }}
+        aria-hidden={!isOpen}
+        transition={{
+          bounce: 0,
+          ease: "circOut",
+        }}
+      >
+        <div className="p-6">
+          <button 
+            onClick={() => setIsOpen(!isOpen)}
```

Comment:
I like to think of this conceptually. Would it make sense for a close button to end up opening the search? Given this, it's better to be explicit
```suggestion
            onClick={() => setIsOpen(false)}
```